ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Examples.hs

```diff
@@ -61,9 +76,10 @@ codecConfig = ShelleyCodecConfig
 
 fromShelleyLedgerExamples
   :: ShelleyBasedEra era
-  => ShelleyLedgerExamples era
+  => Core.TranslationContext era
```

Comment:
This parameter seems off to me. Two ideas:

- Maybe what we actually want is to add a `TranslationContext` upstream to the `ShelleyLedgerExamples` data type?

- In the meantime, we could use a new class here. It is not our intention to test with multiple `TranslationContexts`s -- but since the type varies per era, we can't trivially bake such a context into this definition. Hence this parameter. But since our intentions are all static and vary only wrt to era, we could use a class to emphasize that.

Reply:
@nfrisby Sorry for the delay. I totally agree with your first bullet I can make a PR for that in the cardano-ledger repo.

Your second bullet is simply an alternative to the current implementation, which can also be seen as a temporary solution before we can integrate upstream changes to cardano-ledger, correct? I don't really agree with the use of a class here; it seems over engineered. In particular, if I made a class it would only be used in this one module and only be used once per instance (i.e. per era), so it would be reduced to a complicated way of passing a function argument. We don't gain anything in terms of code reuse and even if did, this would all be soon replaced with the upstream changes. Is this convincing?

Reply:
Any progress with that upstream PR? Looks like nothing has changed in this part of the diff yet.

Reply:
@nfrisby I've made a PR here: https://github.com/input-output-hk/cardano-ledger/pull/2598

Reply:
In the latest [update](https://github.com/input-output-hk/ouroboros-network/compare/b3164abcf9d7790a66514af3390687ab00ce03eb..63095e6f38c282592f0413376fbde06941351fb9), I'm using the upstream example.